packages/subgraph/src/utils/tokens/erc1155.ts

```diff
@@ -0,0 +1,349 @@
+import {Address, BigInt, Bytes, ethereum, log} from '@graphprotocol/graph-ts';
+import {
+  ERC1155Balance,
+  ERC1155Contract,
+  ERC1155TokenIdBalance,
+  ERC1155Transfer
+} from '../../../generated/schema';
+import {ERC1155} from '../../../generated/templates/DaoTemplateV1_0_0/ERC1155';
+import {supportsInterface} from '../erc165';
+import {
+  DECODE_OFFSET,
+  ERC1155_MAGIC_NUMBER,
+  ERC1155_safeBatchTransferFrom,
+  ERC1155_safeTransferFrom,
+  ERC165_MAGIC_NUMBER,
+  TransferType,
+  getBalanceId,
+  getERC1155TransferId,
+  getTokenIdBalanceId
+} from './common';
+
+export function supportsERC1155(token: Address): bool {
+  // Double check that it's ERC1155 by calling supportsInterface checks.
+  let erc1155 = ERC1155.bind(token);
+  let introspection_ERC165 = supportsInterface(erc1155, ERC165_MAGIC_NUMBER); // ERC165
+  let introspection_ERC1155 = supportsInterface(erc1155, ERC1155_MAGIC_NUMBER); // ERC1155
+  let introspection_00000000 = supportsInterface(erc1155, '00000000', false);
```

Comment:
What is this check for?

Reply:
@Rekard0 I think we do this wrong (also for ERC-20).

To comply with ERC-165, implementing contracts must return false for `0xfffff` (see https://eips.ethereum.org/EIPS/eip-165):

> Therefore the implementing contract will have a supportsInterface function that returns:
>
> - true when interfaceID is 0x01ffc9a7 (EIP165 interface)
> - false when interfaceID is 0xffffffff
> - true for any other interfaceID this contract implements
> - false for any other interfaceID